src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/AssemblyReference.cs

```diff
@@ -122,6 +122,11 @@ public BlobHandle HashValue
             }
         }
 
+        public AssemblyName GetAssemblyName()
+        {
+            return _reader.GetAssemblyName();
+        }
```

Comment:
This doesn't seem right. You need to fetch the name details for the current AssemblyReference using the row id and/or other members of this type. Once you do that, I don't think there will be much to share between def and ref and you won't even need the internal method on MetadataReader.